pkg/detectors/sendbird/sendbird.go

```diff
@@ -61,20 +68,42 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 					continue
 				}
 				req.Header.Add("Api-Token", resMatch)
+
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+
 				res, err := client.Do(req)
 				if err == nil {
 					defer res.Body.Close()
 					if res.StatusCode >= 200 && res.StatusCode < 300 {
 						s1.Verified = true
-					} else {
-						// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
-						if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
-							continue
+					} else if res.StatusCode == 400 { // Sendbird returns 400 for all errors
+						var userResp userResp
+						err := json.NewDecoder(res.Body).Decode(&userResp)
+						if err != nil {
+							s1.VerificationError = fmt.Errorf("error decoding json response body: %w", err)
+						}
```

Comment:
It looks like you keep on truckin' even if the JSON couldn't be decoded - was that intentional?

Reply:
yes, i assumed if the json decode failed that would still count as an indeterminate case. do you have a different opinion?

Reply:
i guess we can potentially update the next if condition to be `else if userResp.Code != 400401` such that it doesn't run if decode fails?

Reply:
yeah, that's where i was going. as written right now, you set the error information, but then immediately check `userResp.Code`, which will always be zero, meaning you'll immediately overwrite the error information you just set with `fmt.Errorf("unexpected response code: %d", userResp.Code)`, which is inaccurate.